src/bin/ion/commands/inspect.rs

```diff
@@ -393,26 +499,23 @@ impl<'a, 'b> IonInspector<'a, 'b> {
         comment_fn: impl CommentFn<'x>,
     ) -> Result<()> {
         use ExpandedValueSource::*;
-        let value_literal = match value.expanded().source() {
-            ValueLiteral(value_literal) => value_literal,
-            // In Ion 1.0, there are no template values or constructed values so we can defer
-            // implementing these.
-            Template(_, _) => {
-                todo!("Ion 1.1 template values")
-            }
-            Constructed(_, _) => {
-                todo!("Ion 1.1 constructed values")
-            }
-        };
-
-        use LazyRawValueKind::*;
-        // Check what encoding this is. At the moment, only binary Ion 1.0 is supported.
-        match value_literal.kind() {
-            Binary_1_0(bin_val) => {
-                self.inspect_binary_1_0_scalar(depth, delimiter, value, bin_val, comment_fn)
+        match value.expanded().source() {
+            // If the value is backed by an encoded literal AND that literal wasn't passed as an
+            // argument to an e-expression, inspect the encoded value.
+            ValueLiteral(value_literal) if !value.expanded().is_parameter() => {
+                use LazyRawValueKind::*;
+                match value_literal.kind() {
+                    Binary_1_0(bin_val) => {
+                        self.inspect_literal_scalar(depth, delimiter, value, bin_val, comment_fn)
+                    }
+                    Binary_1_1(bin_val) => {
+                        self.inspect_literal_scalar(depth, delimiter, value, bin_val, comment_fn)
```

Comment:
🗺️ The `inspect_literal_scalar` method is generic over the encoding of the binary value, so these two method calls look identical but they're different.

Reply:
It's more obvious in an IDE, but I though it might raise an eyebrow in the context of the diff.